modules/posts/includes/class-post-list-base.php

```diff
@@ -20,33 +20,63 @@ function __construct( $posts, $args = array() ) {
 
 			$args = wp_parse_args( $args, array(
 				'list_owner' => false,
-			));
+			) );
 
 			/**
 			 * @var WPLib_Posts $list_owner
 			 */
-			$list_owner = $args['list_owner'];
+			$list_owner = $args[ 'list_owner' ];
 
 			if ( $list_owner ) {
 
-				foreach ($posts as $index => $post) {
+				if ( WPLib::can_call( 'make_new_item', $list_owner ) ) {
```

Comment:
@sarutole These are apparently the really important changes.  I am worries about unintended consequences of these.  Can you talk through these changes and discuss why they were made?

FYI, I think these methods are potential candidates for eventual refactoring because of their brittleness.